### ISSUE TITLE
Different behaviour when parsing empty columns

### ISSUE THREAD

**Issue by dominik1001** [tilo/smarter_csv#168]
Version: smarter_csv (1.2.8)
Settings: `{ remove_empty_values: false }`

---

When I have the csv

```
id,col1,col2,col3
123,,,
```

smarter csv parses

```
{ :id => 123,
    :col1 => nil,
    :col2 => nil,
    :col3 => nil}
```

for the first row. But when I have the csv

```
id,col1,col2,col3
123,,,1
```

the result is

```
{:id => 123,
    :col1 => "",
    :col2 => "",
    :col3 => 1}
```

IMO it should be either `nil` or `''` for col1 and col2, but it should not behave differently.
What do you guys think?

Thanks,
Dominik

**Comment by stepanBoichyshyn** [comment]
@dominik1001 This is because `SmarterCSV` replace all nil values by empty string permanently
https://github.com/tilo/smarter_csv/blob/f75c099b8b6d4af6c5496c39ed4901e754672245/lib/smarter_csv/smarter_csv.rb#L199

I currently looking for a solution to return nils anyway. Will ping you with a results

**Comment by tilo** [comment]
it really boils down to some wacky behavior of the String#split method:

```
>  '123,,,'.split(',')
 => ["123"]
 
 >  '123,,,1'.split(',')
 => ["123", "", "", "1"]
 ```

**Comment by tilo** [comment]
@dominik1001 @stepanBoichyshyn fixed in version 1.4.0
